modules/check_in/app/sidekiq/check_in/travel_claim_base_job.rb

```diff
@@ -44,27 +45,49 @@ class TravelClaimBaseJob
                                  Constants::CIE_ERROR_TEMPLATE_ID, Constants::OH_ERROR_TEMPLATE_ID,
                                  Constants::OH_FAILURE_TEMPLATE_ID, Constants::OH_TIMEOUT_TEMPLATE_ID].freeze
 
-    MAX_RETRIES = 3
-
     def send_notification(opts = {})
       log_sending_travel_claim_notification(opts)
-
       retry_attempt = 0
+      if self.class.sidekiq_options_hash['retry_count']
+        retry_attempt = self.class.sidekiq_options_hash['retry_count'].to_i
+      end
+      attempt_number = retry_attempt + 1
+
       begin
         va_notify_send_sms(opts)
       rescue => e
-        retry_attempt += 1
-        if retry_attempt <= MAX_RETRIES
-          log_send_sms_failure(retry_attempt)
-
-          sleep(retry_attempt * 2) # Backoff: 2s, 4s, 6s
-          retry
-        end
-        handle_error(e, opts)
+        log_send_sms_failure(attempt_number)
+
         raise e
       end
     end
 
+    sidekiq_retries_exhausted do |job, ex|
+      CheckIn::TravelClaimBaseJob.handle_error(job, ex)
+    end
+
+    def self.handle_error(job, ex)
+      opts = job['args'].first
+      template_id = opts[:template_id]
+      SentryLogging.log_exception_to_sentry(
+        ex,
+        { phone_number: opts[:mobile_phone].delete('^0-9').last(4), template_id:,
```

Comment:
What if mobile phone is nil here?

Reply:
@LindseySaari thanks for the sharpe eye, this brought up some important discussion that lead to me realizing some other issues with this update so I will be closing this pr and re-opening from a separate branch with significant changes